Tests/StreamChatTests/Controllers/CurrentUserController/CurrentUserController_Tests.swift

```diff
@@ -424,6 +424,21 @@ final class CurrentUserController_Tests: XCTestCase {
         
         // Assert updater is called with correct data
         XCTAssertEqual(env.currentUserUpdater.addDevice_token, expectedToken)
+        XCTAssertEqual(env.currentUserUpdater.addDevice_pushProvider, PushProvider.apn)
+        XCTAssertNotNil(env.currentUserUpdater.addDevice_completion)
+    }
+
+    func test_addDevice_whenPushProviderIsFirebase_callsCurrentUserUpdaterWithCorrectValues() {
+        // Simulate current user
+        env.currentUserObserverItem = .mock(id: .unique)
+
+        let expectedToken = "test".data(using: .utf8)!
+
+        controller.addDevice(token: expectedToken, pushProvider: .firebase)
+
+        // Assert updater is called with correct data
+        XCTAssertEqual(env.currentUserUpdater.addDevice_token, expectedToken)
+        XCTAssertEqual(env.currentUserUpdater.addDevice_pushProvider, PushProvider.firebase)
```

Comment:
I don't think this is very valuable IMO, we are basically just testing String Represensatble API, which we don't need since it is Apple's API. I guess we can also just make PushProvider an enum 🤔 Do we want customers to extend this themselves? I don't think so 🤔